target/linux/realtek/files-5.15/drivers/net/ethernet/rtl838x_eth.c

```diff
@@ -190,7 +190,7 @@ struct rtl838x_eth_priv {
 	u32 lastEvent;
 	u16 rxrings;
 	u16 rxringlen;
-	u8 smi_bus[MAX_PORTS];
+	u32 smi_bus[MAX_PORTS];
```

Comment:
Since it's not in the commit message: why is the width change needed?

Reply:
Idk; birger did this, but I'm pretty sure, because he's using `of_property_read_u32`.

Reply:
Fair enough, although the property is read into an intermediate variable and should be bound checked (it isn't) before it is assigned to the `priv::smi_bus` and `priv::smi_addr` properties, which both fit into a u8.

Reply:
true, but lets just keep it at u32; no needs for checking; as we do the bounds checking elsewhere anyway (SMI_MAX_BUSSES). Also, this is all 'throw-away-code' anyway :)